packages/web3-eth/types/index.d.ts

```diff
@@ -54,10 +54,11 @@ export class Eth extends AbstractWeb3Module {
 
     clearSubscriptions(): Promise<boolean>;
 
-    subscribe(type: 'logs', options?: Logs, callback?: (error: Error, log: Log) => void): Subscription<Log>;
-    subscribe(type: 'syncing', options?: null, callback?: (error: Error, result: any) => void): Subscription<any>;
-    subscribe(type: 'newBlockHeaders', options?: null, callback?: (error: Error, blockHeader: BlockHeader) => void): Subscription<BlockHeader>;
-    subscribe(type: 'pendingTransactions', options?: null, callback?: (error: Error, transactionHash: string) => void): Subscription<string>;
+    subscribe(
```

Comment:
Making this generic actually loses a lot of good typings for these methods. It before use to map a proper typed single response depending on your type string you supply which is the best kinds of types., the less generic in typings like this the better it is. This will make calling this method super hard in TypeScript world, near impossible to get proper compile time errors.